### PR TITLE
Add SRE specific ircbot plugin

### PR DESCRIPTION
**Pull request by wgordon17**
* Allows for controlling channel through a google sheet
* Parses for SNOW ticket numbers and auto-replies with URL

Signed-off-by: Will Gordon <wgordon@redhat.com>

### CODE REVIEW

File: openshift_tools/ircbot/requirements.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 sopel
+git+https://github.com/nithinmurali/pygsheets.git#staging
```
Comment (aweiteka) [inline openshift_tools/ircbot/requirements.txt:2]: why not just `pygsheets`?

Comment (wgordon17) [reply openshift_tools/ircbot/requirements.txt:2]: Without my merged PR (https://github.com/nithinmurali/pygsheets/pull/147)...it spams stdout with useless messages. But it's not a necessity...just annoying =P

File: openshift_tools/ircbot/openshift_sre/openshift_sre.py [hunk 2]
```diff
@@ -0,0 +1,351 @@
+# coding=utf-8
+"""
+openshift_sre.py - Various helper modules for OpenShift Online SRE team
+Copyright © 2017, Will Gordon <wgordon@redhat.com>
+Licensed under the MIT license.
+"""
+import re
+from datetime import datetime as dt
+from sopel import module
```
Comment (aweiteka) [inline openshift_tools/ircbot/openshift_sre/openshift_sre.py]: sopel is not in the jenkins test machine. You'll need to ignore this error with:

```
from sopel import module # pylint: disable=import-error
```

File: openshift_tools/ircbot/openshift_sre/openshift_sre.py [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+    if gsheet_url:
+        match = re.search(GSHEET_URL_REGEX, gsheet_url)
+        groups = match.groupdict()
+        try:
+            gsheet_client = authorize(service_file=bot.config.openshift_sre.google_service_account_file)
+        except ValueError:
+            bot.say('The Service Account JSON file doesn\'t appear to exist. See '
+                    'https://cloud.google.com/storage/docs/authentication#service_accounts for creating a service '
+                    'account. Then move to ' + bot.config.openshift_sre.google_service_account_file, channel)
+            return False
```
Comment (aweiteka) [inline openshift_tools/ircbot/openshift_sre/openshift_sre.py]: I don't think pylint likes this. What are you doing with a False return? `return None` is better I think.

File: openshift_tools/ircbot/openshift_sre/openshift_sre.py [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+                    'I don\'t have access to that spreadsheet({url}). Are you sure '
+                    '{service_account_email} has been invited to that doc?'.format(
+                        url=gsheet_url,
+                        service_account_email=gsheet_client.oauth.service_account_email), channel)
+            elif err.resp['status'] == '404':
+                bot.say('That spreadsheet ({url}) doesn\'t appear to be correct as I\'m getting a 404 for that.'.format(
+                    url=gsheet_url))
+            else:
+                bot.msg(channel, 'Unknown error: ' + str(err))
+            return False
```
Comment (aweiteka) [inline openshift_tools/ircbot/openshift_sre/openshift_sre.py]: `return None`

File: openshift_tools/ircbot/openshift_sre/openshift_sre.py [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+    # Find worksheet based on `gid` in URL
+    for worksheet in sheet.worksheets():
+        if str(worksheet.id) == groups['gid']:
+            current_worksheet = worksheet
+            break
+    else:
+        bot.say('The worksheet (gid={gid}) on that spreadsheet ({title}) doesn\'t seem to exist.'.format(
+            gid=groups['gid'],
+            title=sheet.title), channel)
+        return False
```
Comment (aweiteka) [inline openshift_tools/ircbot/openshift_sre/openshift_sre.py]: `return None`

**Review by aweiteka — CHANGES_REQUESTED** [review]

**Comment by aweiteka** [comment]
[test]

**Comment by aweiteka** [comment]
@aweiteka +1 pr #3098

**Pull request merged**